include/wsrep/streaming_context.hpp

```diff
@@ -204,6 +216,7 @@ namespace wsrep
         size_t fragment_size_;
         size_t unit_counter_;
         size_t log_position_;
+        int sr_store_;
```

Comment:
This variable could perhaps be moved into storage_service implementation on MariaDB side.

Reply:
It can not be moved to `Wsrep_storage_service` on the MariaDB side because the lifetime of a `Wsrep_storage_service` object is too short: the `sr_store` value should have the same lifetime as the transaction, but on the master an `Wsrep_storage_service` object exists only for the duration of the `append_fragment()` call that adds a fragment to the fragment table. In addition to this, on the slave side the `Wsrep_storage_service` objects are not used at all for SR transactions.

But maybe there is some other class on the MariaDB side where `sr_store` could be moved.